### PR TITLE
Add parameters for S3 compatible storage , issue #321 

### PR DESCRIPTION
**Pull request by zz22394**
I added two parameters for issue #321 
1)hive.s3.endpoint
Allow endpoint other than aws official server.
2)hive.s3.signer-type
Enable specifying Sign type when connecting S3 API.
Possible value is [here](https://github.com/elastic/elasticsearch/pull/13360)

### CODE REVIEW

**Comment by amrutagokhale** [comment]
Can you please rewrite the commit message as `Add config properties for S3-compatible storage`?

File: presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java [hunk 1]
```diff
@@ -617,6 +623,18 @@ private AmazonS3Client createAmazonS3Client(URI uri, Configuration hadoopConfig,
...
         }
 
+        String endPoint = hadoopConfig.get(S3_ENDPOINT);
+        if (endPoint != null) {
+            try {
+                client.setEndpoint(endPoint);
+            }
+            catch (IllegalArgumentException e) {
+                String msg = "Incorrect endpoint: "  + e.getMessage();
+                LOG.error(msg);
```
Comment (anusudarsan) [inline presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java]: Exceptions are already logged, so you do not have to log it separately.

**Comment by zz22394** [comment]
@amrutagokhale I changed the commit message.

File: presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java (hunk 1 above)
Comment (zz22394) [reply presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java]: @anusudarsan Removed the unnecessary try-catch and logging.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestPrestoS3FileSystem.java [hunk 2]
```diff
@@ -74,6 +74,23 @@ public void testStaticCredentials()
     }
 
     @Test
+    public void testCompatiableStaticCredentials()
```
Comment (rschlussel-zz) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestPrestoS3FileSystem.java]: typo: should say Compatible

Comment (zz22394) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestPrestoS3FileSystem.java]: @rschlussel Sorry for the typo miss. It is fixed now.

**Comment by mattsfuller** [comment]
This was merged upstream to prestodb/presto. I will close this as this changed is picked up when we update our fork.

**Pull request closed without merging**